text/0113-canonical-metadata.md

```diff
@@ -0,0 +1,625 @@
+                    "items": { 
+                        "$ref": "#/definitions/DynamicComponentReference"
+                    }
+                },
+                "directives": {
+                    "type": "object",
+                    "properties": {
+                        "forEach": {
+                            "type": "array", 
+                            "items": { "$ref" : "#/definitions/TemplateForEachDirective" }
+                        },
+                        "iterator": {
+                            "type": "array", 
+                            "items": { "$ref" : "#/definitions/TemplateIteratorDirective" }
+                        },
+                        "if": {
+                            "type": "array", 
+                            "items": { "$ref" : "#/definitions/TemplateIfDirective" }
+                        },
+                        "lwcDom": {
+                            "type": "array", 
+                            "items": { "$ref" : "#/definitions/TemplateLwcDomDirective" }
+                        }
+                    }
+                }
+            }
+        },
+        "JSFile": {
+            "type": "object",
+            "properties": {
+                "fileType": { "$ref": "#/definitions/FileType" },
+                "fileName": { "type": "string" },
+                "componentClasses": {
+                    "type": "array",
+                    "items": { 
+                        "$ref": "#/definitions/Class"
+                    }
+                },
+                "imports": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/Import"
+                    }
+                },
+                "dynamicImports": {
+                    "type": "array",
+                    "items": { 
+                        "$ref": "#/definitions/DynamicImport"
+                    }
+                },
+                "exports": {
+                    "type": "array",
+                    "items": [
+                        { "$ref": "#/definitions/Export"},
+                        { "$ref": "#/definitions/AggregatingExport"}
+                    ]
+                },
+                "programmaticEvents": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/ProgrammaticEvent"
+                    }
+                },
+                "staticResources": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "#/definitions/StaticResourceReference"
+                    }
+                }
+            }
+        },
+        "CSSFile": {
+            "type": "object",
+            "properties": {
+                "fileType": { "$ref": "#/definitions/FileType" },
+                "fileName": { "type": "string" },
+                "tokens": { "type": "array", "items": { "type": "string" }},
```

Comment:
Let's remove this to `customProperties`. 

On top of listing all the custom properties present in the CSS, it might also be interesting to add the fallback value if present. This would be interesting information to surface.